### PR TITLE
Simfony 4 upgrade

### PR DESCRIPTION
**Pull request by mantas-gudelevicius-hiccup**
- updated to Symfony 4 requirements
- updated to php 7.1.x
- added typehinting

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
     "name": "gusdecool/encryption-bundle",
+    "version": "1.1.0",
```
Comment (gusdecool) [inline composer.json]: remove `version` tag

File: composer.json [hunk 2]
```diff
@@ -15,9 +16,9 @@
         }
     },
     "require": {
-        "php": ">=5.4.0",
-        "gusdecool/cryptography": "~1.0",
-        "doctrine/orm": "~2.2",
-        "sensio/framework-extra-bundle": "~2.3|~3.0"
+        "php": "^7.1.3",
```
Comment (gusdecool) [inline composer.json]: change to `~7.0`

File: composer.json [hunk 3]
```diff
@@ -15,9 +16,9 @@
...
+        "gusdecool/cryptography": "~1.1|dev-mcrypt_to_openssl",
```
Comment (gusdecool) [inline composer.json]: remove `dev-mcrypt_to_openssl`, we should not using `branch` for production usage

File: src/Subscriber/AesSubscriber.php [hunk 4]
```diff
@@ -164,7 +169,7 @@ private function getAnnotatedProperties($entity)
      *
      * @return string|null
      */
-    private function getGetter($property, $entity)
+    private function getGetter(\ReflectionProperty $property, object $entity): ?string
```
Comment (gusdecool) [inline src/Subscriber/AesSubscriber.php]: remove `?string`, as we're targetting `7.0` that not yet suport `?`

File: src/Subscriber/AesSubscriber.php [hunk 5]
```diff
@@ -181,7 +186,7 @@ private function getGetter($property, $entity)
      *
      * @return string|null
      */
-    private function getSetter($property, $entity)
+    private function getSetter(\ReflectionProperty $property, object $entity): ?string
```
Comment (gusdecool) [inline src/Subscriber/AesSubscriber.php]: same here :v:

File: src/Subscriber/AesSubscriber.php [hunk 6]
```diff
@@ -141,8 +145,9 @@ private function encrypt(LifecycleEventArgs $args)
      * @param object $entity
      *
      * @return \ReflectionProperty[]
+     * @throws \ReflectionException
      */
-    private function getAnnotatedProperties($entity)
+    private function getAnnotatedProperties(object $entity): array
```
Comment (gusdecool) [inline src/Subscriber/AesSubscriber.php]: i'm not sure `object` is not valid data type in PHP, can you please point me to this documentation? 🤔

**Review by gusdecool — CHANGES_REQUESTED** [review]
We also need to wait https://github.com/gusdecool/cryptography/pull/1 merged and release before merged this PR 🙏

File: src/Subscriber/AesSubscriber.php (hunk 6 above)
Comment (mantas-gudelevicius-hiccup) [reply src/Subscriber/AesSubscriber.php]: http://php.net/manual/en/migration72.new-features.php it's new in php 7.2

File: composer.json (hunk 3 above)
Comment (gusdecool) [reply composer.json]: you can change this to `~2.0` ✌️

**Review by gusdecool — APPROVED** [review]

**Pull request merged**